crates/llms/src/embeddings/candle/util.rs

```diff
@@ -107,19 +105,58 @@ pub(crate) fn download_hf_artifacts(
     };
     let api_repo = api.repo(repo.clone());
 
-    tracing::trace!("Downloading 'config.json' for {}", repo.url());
+    Ok(api_repo)
+}
+
+pub fn download_hf_file(
+    repo_id: &str,
+    revision: Option<&str>,
+    repo_type_opt: Option<&str>,
+    file: &str,
+    hf_token: Option<&str>,
+) -> Result<PathBuf, Box<dyn std::error::Error + Send + Sync>> {
+    let api = ApiBuilder::new()
+        .with_progress(false)
+        .with_token(hf_token.map(ToString::to_string))
+        .build()
+        .boxed()?;
+
+    let repo_type = match repo_type_opt {
```

Comment:
Better to warn/error if invalid repo_type specified  - fro example typo `dattasets`